src/lib/coda_graphql/coda_graphql.ml

```diff
@@ -456,16 +456,15 @@ module Types = struct
 
   module Next_epoch_data =
     Make_epoch_data (struct
-        type t = State_hash.t option
```

Comment:
NIT: Since next_epoch_data and staking_epoch_data are the same type now, we can make the Make_epoch_data functor a lot simpler. In fact, I think we can replace the functor with a function that creates the epoch data type, where the inputs take a name parameter and a first class module. the first class module interface would be something like the following:

```
module type Intf = Consensus.Epoch_data_intf
                with type ledger := Consensus.Data.Epoch_ledger.Value.t
                 and type seed := Consensus.Data.Epoch_seed.t
                 and type lock_checkpoint := State_hash.t
                 and type Value.t := Lock_checkpoint.epoch_data
```

Can you make this refactoring before landing your PR?

Reply:
:+1: